lib/ocpp/common/websocket/websocket_tls_tpm.cpp

```diff
@@ -320,10 +324,11 @@ void WebsocketTlsTPM::recv_loop() {
 }
 
 void WebsocketTlsTPM::client_loop() {
+    conn_data_mutex.lock();
```

Comment:
Question: instead of using manually lock/unlock can we define it in a local scope? The local data can be defined outside that mentioned scope.

Reply:
I've re-read the C++ spec on shared_ptr and thread safety and the mutex can be removed providing that all access is via a separate shared pointer.
i.e. the following would have problems since `conn_data` could change between the nullptr check and the call to do_interrupt()
```
        if (conn_data != nullptr) {
            // another thread could call conn_data.reset() and 
            // the next line would dereference nullptr
            conn_data->do_interrupt();
        }
        conn_data.reset();
```
This is fine since the check is on a separate shared_ptr that can't be reset between operations
```
        std::shared_ptr<ConnectionData> local_data = conn_data;
        if (local_data != nullptr) {
            // another thread could call conn_data.reset() which doesn't 
            // impact local_data other than the reference count
            local_data->do_interrupt();
        }
        conn_data.reset();
```

Updating the code with the new approach